pyqt_sql_demo/connection/model.py

```diff
@@ -30,18 +35,42 @@ def __init__(self, parent):
         self._row_count = 0
         self._column_count = 0
 
-    def connect(self, connection_string):
+    def connect(self, database_type, connection_string):
         # Disconnect from old connection, if any
         self.disconnect()
-        # Strip connection string of missed whitespaces
-        self.attempted_url = connection_string.strip()
-        # Throw error if url is invalid
-        self.verify_attempted_url()
-        # Attempt connection
-        self.con = sqlite3.connect(self.attempted_url)
-        # Use highly-optimized RowFactory to enable
-        # name based access to columns in rows
-        self.con.row_factory = sqlite3.Row
+        self.database_type = database_type
+        if database_type == "SQLite":            
+            # Strip connection string of missed whitespaces
+            self.attempted_url = connection_string.strip()
+            # Throw error if url is invalid
+            self.verify_attempted_url()
+            # Attempt connection
+            self.con = sqlite3.connect(self.attempted_url)
+            # Use highly-optimized RowFactory to enable
+            # name based access to columns in rows
+            self.con.row_factory = sqlite3.Row
+        elif database_type == "MySQL":
+            #host=localhost;port=3307;dbname=mysql.user
+            #connection_string = "username=root;host=localhost;port=3306;dbname=mysql"
+            connection_args = dict(s.split("=") for s in connection_string.split(";"))
+
+            if "dbname" in connection_args:
+                connection_args["database"] = connection_args.pop("dbname")
+
+            if "username" in connection_args:
+                connection_args["user"] = connection_args.pop("username")
```

Comment:
Is there a reason we can't directly put `user` and `database` into the connection string instead of `dbname` and `username` and avoid this conversion all together?